smt/applications/tests/test_podi.py

```diff
@@ -159,27 +207,91 @@ def test_set_options(self):
         """Tests the method that sets the interpolations options settings."""
 
         sm = PODI()
-        sm.compute_pod(self.database, n_modes=5)
-        options = [
+
+        try:
+            sm.set_interp_options()
+        except RuntimeError:
+            pass
+        else:
+            raise RuntimeError(
+                "It should not be possible to call 'set_interp_options' before 'compute_pod'."
+            )
+
+        sm.compute_pod(self.database, n_modes=2)
+
+        try:
+            sm.set_interp_options("non existing surrogate model")
+        except ValueError:
+            pass
+        else:
+            raise RuntimeError(
+                "It should not be possible to initialize an unavailable surrogate model."
+            )
+
+        try:
+            sm.set_interp_options("KRG", [{}, {}, {}])
+        except ValueError:
+            pass
+        else:
+            raise RuntimeError(
+                "It should not be possible to use a non-valid size for the list of options."
+            )
+
+        options_global = [
             {
                 "poly": "quadratic",
                 "corr": "matern32",
                 "pow_exp_power": 0.38,
                 "theta0": [1e-1],
             }
         ]
-        sm.set_interp_options("KRG", options)
+        sm.set_interp_options("KRG", options_global)
 
         sm_list = sm.get_interp_coef()
         for interp_coeff in sm_list:
-            for key in options[0].keys():
-                assert interp_coeff.options[key] == options[0][key]
+            for key in options_global[0].keys():
+                assert interp_coeff.options[key] == options_global[0][key]
+
+        options_local = [{"poly": "quadratic"}, {"corr": "matern52"}]
+        sm.set_interp_options("KRG", options_local)
+
+        sm_list = sm.get_interp_coef()
+        for i, interp_coeff in enumerate(sm_list):
+            for key in options_local[i].keys():
+                assert interp_coeff.options[key] == options_local[i][key]
```

Comment:
Did not notice it before, but you should use assertion methods of `unittest.TestCase` instead of raw `assert`. See [`unittest`](https://docs.python.org/3/library/unittest.html).